snapshot-labs/sx-monorepo
# feat: add notification page

### Summary



Closes: #315

### Changes

- Add a /notifications page, showing activities (proposals start and end) from follow list
- New activities will be marked as `unread`
- Once loaded and /notifications page has been loaded, all current notifications will be marked as `read` when closing the notifications page
- Notifications are refreshed in the background every 15min
- Notifications are reloaded on follow list update
- Load notifications for activities on all networks (need to follow some onchain spaces, waiting for #293)
- Add a new "Notifications" link in the "My" sidebar (with unread notifications count)

### How to test

1. Log in with an account with some followed spaces
2. Go to /notifications
3. You should see start and end activities from your followed spaced
4. You should see the unread count and link to /notifications in the sidebar
5. You should see the unread count in the page title
6. Unread notifications should have a light grey background
7. Navigate to some other page, from the Nav sidebar (the notifications count should disappear)
8. Navigate back to /notifications
9. All notifications should now be marked as read (transparent background)
10. To test the auto-refresh, edit the `REFRESH_INTERVAL` in the notifications store to something smallet (like 15s)
11. Open the dev console, you should see new requests to graphql proposals endpoint on each refresh (2 for each network)
12. Switch wallet, it should refresh the notifications content

Notifications should be marked as all read only once the /notifications page is visited.
